lib/OpenQA/CacheService/Model/Cache.pm

```diff
@@ -48,166 +48,132 @@ sub from_worker {
+            ++$att and sleep 1 and $log->warn("Updating cache failed (attempt $att)")
               until ($ok = $self->_update_asset($asset, $etag, $size)) || $att > 5;
-            unless ($ok) {
-                $log->error('CACHE: FAIL Could not update DB - purging asset');
+
+            if ($ok) { $log->info("Download of $asset successful, new cache size is $self->{cache_real_size}") }
+            else {
+                $log->error("Purging $asset because updating the cache failed, this should never happen");
                 $self->purge_asset($asset);
-                $asset = undef;
             }
-            $log->debug("CACHE: Asset download successful to $asset, Cache size is: $self->{cache_real_size}") if $ok;
         }
         else {
-            $log->debug(
-                "CACHE: Size of $asset differs, Expected: " . $headers->content_length . " / Downloaded: " . "$size");
-            $asset = 598;    # 598 (Informal convention) Network read timeout error
+            $log->warn("Size of $asset differs, expected " . $headers->content_length . " but downloaded $size");
+            $ret = 598;    # 598 (Informal convention) Network read timeout error
         }
     }
     else {
-        my $message = $tx->res->error->{message};
-        $log->debug("CACHE: Download of $asset failed with: $code - $message");
+        my $message = $res->error->{message};
+        $log->warn("Purging $asset because the download failed: $code - $message");
         $self->purge_asset($asset);
-        $asset = undef;
     }
-    return $asset;
+
+    return $ret;
 }
 
 sub get_asset {
     my ($self, $job, $asset_type, $asset) = @_;
 
+    my $log = $self->log;
+
     my $location = path($self->location, base_host($self->host));
     $location->make_path unless -d $location;
     $asset = $location->child(path($asset)->basename);
-    my $log = $self->log;
 
     my $n = 5;
     while (1) {
         $self->track_asset($asset);    # Track asset - make sure it's in DB
         my $result = $self->_asset($asset);
+
         my $ret;
-        eval {
-            $ret
-              = $self->_download_asset($job->{id}, lc($asset_type), $asset,
-                ($result->{etag}) ? $result->{etag} : undef);
-        };
-        if (!$ret) {
-            $asset = undef;
-            last;
-        }
-        elsif ($ret =~ /^5[0-9]{2}$/ && --$n) {
-            $log->debug("CACHE: Error $ret, retrying download for $n more tries");
-            $log->debug("CACHE: Waiting " . $self->sleep_time . " seconds for the next retry");
+        eval { $ret = $self->_download_asset($job->{id}, lc($asset_type), $asset, $result->{etag}) };
 
-            sleep $self->sleep_time;
+        if (!$ret) { last }
+        elsif ($ret =~ /^5[0-9]{2}$/ && --$n) {
+            my $time = $self->sleep_time;
+            $log->warn("Download error $ret, waiting $time seconds for next try ($n remaining)");
```

Comment:
same as above, you consider this a situation we need to do something about? I mean, it already retries and a download failing is something I would foresee whenever network is involved.

Reply:
That's why it's just a warning and not an error.